File: reference/src/adapters/resolve-secret.ts
/**
 * Resolves a secretRef to an environment variable value.
 *
 * Converts kebab-case secret names to UPPER_SNAKE_CASE environment variable
 * names (e.g., `github-token` → `GITHUB_TOKEN`).
 */

export function resolveSecret(secretRef: string): string {
  const envVar = secretRef.replace(/-/g, '_').toUpperCase();
  const value = process.env[envVar];
  if (!value) {
    throw new Error(`Secret "${secretRef}" not found: environment variable ${envVar} is not set`);
  }
  return value;
}

File: sdk-python/src/ai_sdlc/adapters/resolve_secret.py
"""Resolve a secretRef to an environment variable value."""

from __future__ import annotations

import os


def resolve_secret(secret_ref: str) -> str:
    """Convert kebab-case secret name to UPPER_SNAKE_CASE env var and read it.

    Raises ``ValueError`` if the env var is not set.
    """
    env_var = secret_ref.replace("-", "_").upper()
    value = os.environ.get(env_var)
    if not value:
        msg = (
            f'Secret "{secret_ref}" not found: '
            f"environment variable {env_var} is not set"
        )
        raise ValueError(msg)
    return value
